state/runtime/evm/instructions.go

```diff
@@ -1050,19 +958,22 @@ func opPush0(c *state) {
 		return
 	}
 
-	c.push(zero)
+	c.push(*uint256.NewInt(0))
 }
 
 func opPush(n int) instruction {
 	return func(c *state) {
 		ins := c.code
 		ip := c.ip
 
-		v := c.push1()
 		if ip+1+n > len(ins) {
-			v.SetBytes(append(ins[ip+1:], make([]byte, n)...))
+			d := uint256.NewInt(0)
```

Comment:
Move `d := uint256.NewInt(0)` before if condition and `c.push(*d)` after if/else to shorten code.